dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ChatHistoryExtensions.cs

```diff
@@ -25,40 +24,24 @@ public static class ChatHistoryExtensions
     public static async IAsyncEnumerable<StreamingChatMessageContent> AddStreamingMessageAsync(this ChatHistory chatHistory, IAsyncEnumerable<OpenAIStreamingChatMessageContent> streamingMessageContents)
     {
         List<StreamingChatMessageContent> messageContents = new();
+
         // Stream the response.
-        StringBuilder contentBuilder = new();
-        List<ChatCompletionsFunctionToolCall>? functionCallResponses = null;
+        StringBuilder? contentBuilder = null;
+        Dictionary<int, string>? toolCallIdsByIndex = null;
+        Dictionary<int, string>? functionNamesByIndex = null;
+        Dictionary<int, StringBuilder>? functionArgumentBuildersByIndex = null;
         Dictionary<string, object?>? metadata = null;
         AuthorRole? streamedRole = default;
         await foreach (var chatMessage in streamingMessageContents.ConfigureAwait(false))
         {
-            if (metadata is null && chatMessage.Metadata is not null)
-            {
-                metadata = (Dictionary<string, object?>)chatMessage.Metadata;
-            }
+            metadata ??= (Dictionary<string, object?>?)chatMessage.Metadata;
 
             if (chatMessage.Content is { Length: > 0 } contentUpdate)
             {
-                contentBuilder.Append(contentUpdate);
+                (contentBuilder ??= new()).Append(contentUpdate);
```

Comment:
I think majority 99.9% of our cases we will have a streaming with content, and checking if the contentBuilder is null might not be necessary.

Reply:
Majority, yes. When there are function calls, though, it's often empty when not using auto invoke.